plugin/core/sessions.py

```diff
@@ -1104,6 +1125,11 @@ def send_request(
         """You can call this method from any thread. Callbacks will run in Sublime's worker thread."""
         sublime.set_timeout_async(functools.partial(self.send_request_async, request, on_result, on_error))
 
+    def request_promise_async(self, request: Request) -> Promise:
```

Comment:
The naming feels weird as it can be understood as "request a promise" while it's sending a request and returning a promise. Also, the `_async` suffix is confusing due to the pattern that we are using to mark methods that should be called from an async thread.

`send_request_promise` sounds stupid too, as well as `send_request_p`... ;)

Maybe `send_request_task` since it uses a concept of a task internally?

Reply:
renamed to send_request_task